src/JSChallenges/jsArrays.js

```diff
@@ -0,0 +1,420 @@
+myFunction(['test', 'test', 'test'])
+Expected
+true 
+myFunction([1,1,1,2])
+Expected
+false 
+
+//Question 14
+// Write a function that takes arguments an arbitrary number of arrays
+// It should return an array containing the values of all arrays
+function myFunction(...arrays)
+{
+  return arrays.flat()
+}
+//Answer
+myFunction([1, 2, 3], [4, 5, 6])
+Expected
+[1, 2, 3, 4, 5, 6]
+
+//Question 15
+// Write a function that takes an array of numbers as argument
+// Return the number of negative values in the array
+function myFunction(a)
+{
+let result=[]
+for(let i=0;i<a.length;i++){
+if(a[i]<0){
+  result.push(i)
+}
+}
+c=result.length
+return c
+}
+//Another
+function myFunction(a) {
+  return a.filter((el) => el < 0).length;
+}
+//Answer
+myFunction([1,-2,2,-4])
+Expected
+2
+
+//Question 16
+// Write a function that takes two numbers (min and max) as arguments
+// Return an array of numbers in the range min to max
+function myFunction(min, max)
+{
+  return range = [...Array(max- min+ 1).keys()].map(x => x + min);
+}
+//Another
+function myFunction(min, max) {
+  let arr = [];
+  for (let i = min; i <= max; i++) {
+    arr.push(i);
+  }
+return arr;
+}
+//Answer
+myFunction(2, 10)
+Expected
+[2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+//Question 17
+// Write a function that takes an array of strings as argument
+// Group those strings by their first letter
+// Return an object that contains properties with keys representing first letters
+// The values should be arrays of strings containing only the corresponding strings
+// For example, the array ['Alf', 'Alice', 'Ben'] should be transformed to
+// { a: ['Alf', 'Alice'], b: ['Ben']}
+function myFunction(arr)
+{
+let resultObj = {};
+  for (let i =0; i < arr.length; i++) {
+    let currentWord = arr[i];
+    let firstChar = currentWord[0].toLowerCase();
+    let innerArr = [];
+    if (resultObj[firstChar] === undefined) {
+       innerArr.push(currentWord);
+      resultObj[firstChar] = innerArr
+    }else {
+      resultObj[firstChar].push(currentWord)
+    }
+  }
+  return resultObj
+}
+//Another
+function myFunction(arr) {
+  return arr.reduce((acc, cur) => {
+    const firstLetter = cur.toLowerCase().charAt(0);
+    return { ...acc, [firstLetter]: [...(acc[firstLetter] || []), cur] };
+  }, {});
+ }
+//Answer
+myFunction(['Alf', 'Alice', 'Ben'])
+Expected
+{ a: ['Alf', 'Alice'], b: ['Ben']}
+
+//Question 18
+// Write a function that takes two arrays as arguments
+// Merge both arrays and remove duplicate values
+// Sort the merge result in ascending order
+// Return the resulting array
+function myFunction(
+a, b
+)
+{
+let c=a.concat(b)
+ var removeDuplicate= [];
+ for(i=0; i < c.length; i++){
+            if(removeDuplicate.indexOf(c[i]) === -1) {
+                removeDuplicate.push(c[i]);
+            }
+ }
+removeDuplicate.sort(function(a, b){return a-b});
+return removeDuplicate
+}
+//Another
+function myFunction(a, b) {
+  return [...new Set([...a, ...b])].sort((x, y) => x - y);
+}
```

Comment:
nice but does it return a set or array 
why ?

Reply:
It returns array. Merge tow arrays and removed duplicate elements. Then return single array